MoEPluginBase/Manager/MoEPluginController.m

```diff
@@ -0,0 +1,530 @@
+//
+//  MoEPluginController.m
+//  MoEPluginBase
+//
+//  Created by Rakshitha on 09/12/21.
+//
+
+#import <Foundation/Foundation.h>
+#import "MoEPluginController.h"
+#import "MoEPluginMessage.h"
+#import "MoEPluginUtils.h"
+#import "MoEngage/MoEngage.h"
+#import "MOInAppCampaign+Utility.h"
+#import "MoEPluginConstants.h"
+#import "MoEPluginMessageQueueHandler.h"
+#import "MoEngage/MoEngage.h"
+#import "MoEPluginCoordinator.h"
+#import <MoEngageObjCUtils/MoEngageObjCUtils.h>
+#import <MOEngageInApps/MOInApp.h>
+#import <MoEngageMessaging/MoEngageMessaging-Swift.h>
+#import "MOInAppSelfHandledCampaign+Utility.h"
+
+
+@interface MoEPluginController()<MOInAppNativDelegate, MOMessagingDelegate>
+@property(nonatomic, strong) MoEPluginMessageQueueHandler* messageHandler;
+
+
+@end
+
+@implementation MoEPluginController
+
+- (instancetype)init {
+    self.messageHandler = [[MoEPluginMessageQueueHandler alloc]init];
+    return self;
+}
+
+-(void) flushMessageQueue {
+    [self.messageHandler flushMessageQueue];
+}
+
+-(void)queueMessage:(MoEPluginMessage*)message {
+    [self.messageHandler queueMessage:message];
+}
+
+- (void)initializeSDKWithConfig:(MOSDKConfig*)sdkConfig isDefaultInstance:(BOOL)isDefault andLaunchOptions:(NSDictionary*)launchOptions {
+    self.isSDKInitialized = YES;
+    [self setupSDKWithLaunchOptions:sdkConfig isDefaultInstance:isDefault launchOptions:launchOptions];
+}
+
+- (void)initializeDefaultSDKWithConfig:(MOSDKConfig*)sdkConfig withSDKState:(BOOL)isSdkEnabled andLaunchOptions:(NSDictionary*)launchOptions{
+    
+    [self handleSDKState:isSdkEnabled];
+
+    [self initializeSDKWithConfig:sdkConfig isDefaultInstance:true andLaunchOptions:launchOptions];
+}
+
+- (void)handleSDKState:(BOOL)isSdkEnabled {
+    if (isSdkEnabled) {
+        [[MoEngage sharedInstance] enableSDK];
```

Comment:
This should be instance-specific, this will always change the state of default instance.